projects/graph/graph.py

```diff
@@ -7,35 +7,107 @@ class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
         self.vertices = {}
+
+
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
-        pass  # TODO
+        self.vertices[vertex] = set()
+
+
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
-        pass  # TODO
+        if v1 not in self.vertices:
```

Comment:
I wouldn't have the add edge method creating verticies if they do not exist. This should be strictly just creating an edge between them. So I would remove the `add` part and raise an error if they do not exist